### PR TITLE
Feature/233 grids

### LINKED ISSUE #233: grid components
Rolling the following issues into one as they will live in one module:
- #40 
- #41  
- #42  
- #43

### PR DESCRIPTION
**Pull request by zweishar**
Fixes #233

### CODE REVIEW

**Comment by zweishar** [comment]
I will resolve this merge conflict locally after approval.

File: docroot/modules/custom/bos_core/bos_core.module [hunk 1]
```diff
@@ -24,25 +24,6 @@ function bos_core_inline_entity_form_entity_form_alter(&$entity_form, FormStateI
   $entity_form['revision_log_message']['#access'] = FALSE;
 }
 
-/**
- * Implements hook_preprocess_HOOK().
- */
-function bos_core_preprocess_field(&$variables) {
```
Comment (davidrkupton) [inline docroot/modules/custom/bos_core/bos_core.module]: I am not sure if I put this in, but like this.
I have just removed all the code in this hook on my branch and changed to:
```
  if (!empty($variables['items']) && count($variables["items"][0]["attributes"]->storage()) == 0) {
    unset($variables["items"][0]['attributes']);
  }
```
Which on my branch catches the dread "attributes missing" error with our templating.
I am happy to debate if this should be in the `theme` or in this `bos_core`, but I think it has value as a "safety".

Comment (davidrkupton) [reply docroot/modules/custom/bos_core/bos_core.module]: OK, so I have moved this into the theme in my branch.

Comment (zweishar) [reply docroot/modules/custom/bos_core/bos_core.module]: @davidrkupton I refactored the template that was causing this issue https://github.com/CityOfBoston/boston.gov-d8/pull/237/commits/80d1b2cbbcd3153f8466fea220ccb574c8555276

So I removed that code as we shouldn't need it any longer.

**Pull request merged**